pkg/client/cloudevents/bundle_codec.go

```diff
@@ -81,12 +79,15 @@ func (codec *BundleCodec) Decode(evt *cloudevents.Event) (*api.Resource, error)
 		return nil, fmt.Errorf("failed to convert cloudevent to resource status: %v", err)
 	}
 
+	if originalSource != codec.sourceID {
+		return nil, fmt.Errorf("unmatched original source id %s for resource %s", originalSource, resourceID)
+	}
```

Comment:
what is the case for this mismatch? Do we need to throw error or just skip?

Reply:
in normal case, the original source should always be equal the source,  I think this check can help to prevent the  unexpected error, like we check if the resource id exists